app/actions/handlers.py

```diff
@@ -138,113 +139,126 @@ async def action_pull_events(integration, action_config: PullEventsConfig):
-                                    extra={
-                                        'needs_attention': True,
-                                        'integration_id': str(integration.id),
-                                        "aoi": aoi,
-                                        'action_id': "pull_events"
-                                    }
-                                )
-                                response_per_aoi.append({"aoi": aoi, "response": []})
-                                continue
-                            else:
-                                # Update states
-                                state = {
-                                    "start_time": transformed_data[0].get("recorded_at")
-                                }
-                                await state_manager.set_state(
-                                    str(integration.id),
-                                    "pull_events",
-                                    state,
-                                    aoi
-                                )
-                                response_per_aoi.append({"aoi": aoi, "response": response})
-                else:
-                    response_per_aoi.append({"aoi": aoi, "response": []})
-
-            if events_to_patch:
-                # Process events to patch
-                response = await patch_events(events_to_patch, updated_config_data, integration)
-                result["events_updated"] = len(events_to_patch)
-                result["details"]["updated"] = response
     except httpx.HTTPError as e:
-        message = f"pull_observations action returned error. Integration: {str(integration.id)}. Exception: {e}"
-        logger.exception(message, extra={
+        msg = f"pull_observations action returned error. Integration: {str(integration.id)}. Exception: {e}"
+        logger.exception(msg, extra={
             "integration_id": str(integration.id),
             "attention_needed": True
         })
         raise e
     else:
-        result["events_extracted"] = total_events
-        result["details"]["created"] = response_per_aoi
+        if all([len(items) == 0 for items in events.values()]):
+            logger.info(f"No events were pulled for integration: '{str(integration.id)}'.")
+            result["message"] = f"No events were pulled for integration: '{str(integration.id)}'."
+            return result
+
+        event_ids = []
+        async def get_skylight_events_to_patch():
+            # Get through the events and check if state_manager has it recorded from a previous execution
+            patch_these_events = []
+            for aoi, events_list in events.items():
+                for event in events_list:
+                    event_id = get_clean_event_id(event)
+                    event_ids.append(event_id)
+                    saved_event = await state_manager.get_state(str(integration.id), "pull_events", event_id)
+                    if saved_event:
```

Comment:
The walrus operator can be useful in cases like this:
```suggestion
                    if saved_event := await state_manager.get_state(str(integration.id), "pull_events", event_id)
```